network/manager.go

```diff
@@ -475,39 +475,52 @@ func (nm *networkManager) DeleteEndpoint(networkID, endpointID string, epInfo *E
 		return err
 	}
 
-	err = nm.save()
-	if err != nil {
-		return err
-	}
-
 	return nil
 }
 
 func (nm *networkManager) DeleteEndpointState(networkID string, epInfo *EndpointInfo) error {
 	nw := &network{
-		Id:           networkID,
+		Id:           networkID, // CHECK: currently unused in stateless cni
+		HnsId:        epInfo.HNSNetworkID,
 		Mode:         opModeTransparentVlan,
 		SnatBridgeIP: "",
+		NetNs:        dummyGUID, // to trigger hns v2, windows
 		extIf: &externalInterface{
 			Name:       InfraInterfaceName,
 			MacAddress: nil,
 		},
 	}
 
 	ep := &endpoint{
-		Id:                       epInfo.Id,
+		Id:                       epInfo.EndpointID,
 		HnsId:                    epInfo.HNSEndpointID,
-		HostIfName:               epInfo.IfName,
+		HNSNetworkID:             epInfo.HNSNetworkID, // unused (we use nw.HnsId for deleting the network)
+		HostIfName:               epInfo.HostIfName,
 		LocalIP:                  "",
 		VlanID:                   0,
-		AllowInboundFromHostToNC: false,
+		AllowInboundFromHostToNC: false, // stateless currently does not support apipa
 		AllowInboundFromNCToHost: false,
 		EnableSnatOnHost:         false,
 		EnableMultitenancy:       false,
-		NetworkContainerID:       epInfo.Id,
+		NetworkContainerID:       epInfo.NetworkContainerID, // we don't use this as long as AllowInboundFromHostToNC and AllowInboundFromNCToHost are false
+		NetNs:                    dummyGUID,                 // to trigger hnsv2, windows
+		NICType:                  epInfo.NICType,
+		NICName:                  epInfo.IfName,
```

Comment:
ADD a comment in here that this filed will be used for deleting the delegated NIC in case of Swift V2

Reply:
this is the comment we were discussing earlier regarding what changes we want to make if we need stateless cni to work with linux in the future, right?